engine/src/main/scala/cromwell/engine/workflow/lifecycle/execution/ExecutionStore.scala

```diff
@@ -44,94 +46,81 @@ case class ExecutionStore(private val store: Map[JobKey, ExecutionStatus]) {
       case node: GraphNode => node.upstreamAncestry exists hasFailedScope
     }
 
-    store exists {
-      case (jobKey, jobStatus) => (jobStatus == NotStarted && !upstreamFailed(jobKey.scope)) || jobStatus == QueuedInCromwell || jobStatus == Starting || jobStatus == Running
-    }
+    keysWithStatus(NotStarted).exists(jobKey => !upstreamFailed(jobKey.scope)) ||
+      keysWithStatus(QueuedInCromwell).nonEmpty ||
+      keysWithStatus(Starting).nonEmpty ||
+      keysWithStatus(Running).nonEmpty
   }
 
-  def jobStatus(jobKey: JobKey): Option[ExecutionStatus] = store.get(jobKey)
+  def jobStatus(jobKey: JobKey): Option[ExecutionStatus] = statusStore.get(jobKey)
 
-  def unstartedJobs: List[BackendJobDescriptorKey] = {
-    store.collect({case (k: BackendJobDescriptorKey, status) if status != ExecutionStatus.NotStarted => k }).toList
+  def startedJobs: List[BackendJobDescriptorKey] = {
+    store.filterNot(_ == NotStarted).values.toList.flatten collect {
+      case k: BackendJobDescriptorKey => k
+    }
   }
 
-  private def hasFailedScope(s: GraphNode): Boolean = {
-    store.exists({ case (key, status) => status == Failed && key.scope == s })
-  }
+  private def hasFailedScope(s: GraphNode): Boolean = keysWithStatus(Failed).exists(_.scope == s)
 
-  def hasFailedJob: Boolean = {
-    store.values.exists(_ == ExecutionStatus.Failed)
-  }
+  def hasFailedJob: Boolean = keysWithStatus(Failed).nonEmpty
 
   override def toString = store.map { case (j, s) => s"$j -> $s" } mkString System.lineSeparator()
 
-  def add(values: Map[JobKey, ExecutionStatus]) = this.copy(store = store ++ values)
+  def add(values: Map[JobKey, ExecutionStatus]) = {
+    val hasTerminalStatus = values.values.exists(_.isTerminal)
```

Comment:
what about `lazy val`ing this so it isn't evaluated if `hasNewRunnables` is `true`?